eng/restore-toolset.ps1

```diff
@@ -133,4 +133,39 @@ function InstallDotNetSharedFramework([string]$version) {
   }
 }
 
+# Let's clear out the stage-zero folders that map to the current runtime to keep stage 2 clean
+function CleanOutStage0ToolsetsAndRuntimes {
+  $GlobalJson = Get-Content -Raw -Path (Join-Path $RepoRoot 'global.json') | ConvertFrom-Json
+  $dotnetSdkVersion = $GlobalJson.tools.dotnet
+  $dotnetRoot = $env:DOTNET_INSTALL_DIR
+  $versionPath = Join-Path $dotnetRoot '.version'
+  $aspnetRuntimePath = [IO.Path]::Combine( $dotnetRoot, 'shared' ,'Microsoft.AspNetCore.App')
+  $coreRuntimePath = [IO.Path]::Combine( $dotnetRoot, 'shared' ,'Microsoft.NETCore.App')
+  $wdRuntimePath = [IO.Path]::Combine( $dotnetRoot, 'shared', 'Microsoft.WindowsDesktop.App')
+  $sdkPath = Join-Path $dotnetRoot 'sdk'
+  $majorVersion = $dotnetSdkVersion.Substring(0,1)
+
+  if (Test-Path($versionPath)) {
+    $lastInstalledSDK = Get-Content -Raw -Path ($versionPath)
+    if ($lastInstalledSDK -ne $dotnetSdkVersion)
+    {
+      $dotnetSdkVersion | Out-File -FilePath $versionPath -NoNewline
+      Remove-Item (Join-Path $aspnetRuntimePath "$majorVersion.*") -Recurse
+      Remove-Item (Join-Path $coreRuntimePath "$majorVersion.*") -Recurse
+      Remove-Item (Join-Path $wdRuntimePath "$majorVersion.*") -Recurse
+      Remove-Item (Join-Path $sdkPath "$majorVersion.*") -Recurse
+      Remove-Item (Join-Path $dotnetRoot "packs") -Recurse
+      Remove-Item (Join-Path $dotnetRoot "sdk-manifests") -Recurse
+      Remove-Item (Join-Path $dotnetRoot "templates") -Recurse
+      throw "Installed a new SDK, deleting existing shared frameworks and sdk folders. Please rerun build"
+    }
+  }
+  else
+  {
+    $dotnetSdkVersion | Out-File -FilePath $versionPath -NoNewline
+  }
+}
```

Comment:
So, if there is a `$versionPath` but the `$lastInstalledSDK` is the same as the SDK version in the global.json, this function is a no-op, correct? And if either there is no `$versionPath` or there is one and `$lastInstalledSDK` isn't the same version in the global.json, then it will write out the version into `$versionPath`, correct?

Reply:
correct. The logic above is if the file doesn't exist, write out the version from global.json (this is basically so it doesn't fail the first build run in any new repo). If the file does exist AND it's different than what's in global.json, delete everything, write out the file, and exit.

So it should never write to the file if it matches what's in global.json